samples/materialize_workbooks.py

```diff
@@ -0,0 +1,106 @@
+                        help='enable/disable materialized views for sites/workbooks')
+    parser.add_argument('--status', '-st', required=False, action='store_true',
+                        help='show materialized views enabled sites/workbooks')
+    parser.add_argument('--site-id', '-si', required=False,
+                        help='set to Default site by default')
+    parser.add_argument('--logging-level', '-l', choices=['debug', 'info', 'error'], default='error',
+                        help='desired logging level (set to error by default)')
+    parser.add_argument('--type', '-t', required=False, choices=['site', 'workbook'],
+                        help='type of content you want to update materialized views settings on')
+
+    args = parser.parse_args()
+
+    if args.password:
+        password = args.password
+    else:
+        password = getpass.getpass("Password: ")
+
+    logging_level = getattr(logging, args.logging_level.upper())
+    logging.basicConfig(level=logging_level)
+
+    # site content url is the TSC term for site id
+    site_content_url = args.site_id if args.site_id is not None else ""
+    enable_materialized_views = args.mode == "enable"
+
+    if args.type is not None and args.mode is None:
+        print("Use '--mode <enable/disable>' to specify how you want to change materialized views settings.")
+        return
+
+    if args.type is None and args.mode is not None:
+        print("Use '--type <workbook/site>' to specify the level of materialized views setting you want to change.")
+        return
+
+    if args.type is None and args.mode is None and args.status is None:
+        print("Use '--type <workbook/site> --mode <enable/disable>' to change materialized views settings.")
+        print("Or use '--status' to show currently materialized views enabled workbooks and sites.")
+        return
+
+    if args.type is not None and enable_materialized_views is not None:
+        if args.type == 'site':
+            tableau_auth = TSC.TableauAuth(args.username, password, site_id=site_content_url)
+            server = TSC.Server(args.server)
+
+            with server.auth.sign_in(tableau_auth):
+                site_to_update = server.sites.get_by_content_url(site_content_url)
+                site_to_update.materialized_views_enabled = enable_materialized_views
+
+                server.sites.update(site_to_update)
+                print "Updated materialized views settings for site:", site_to_update.name
+            print
+
+        elif args.type == 'workbook':
+            tableau_auth = TSC.TableauAuth(args.username, password, site_id=site_content_url)
+            server = TSC.Server(args.server)
+
+            # Now it updates all the workbooks in the site
+            # To update selected ones pass filter into 'workbooks = server.workbooks.get()[0]'
+            # This only updates the workbooks in the site you are signing into
```

Comment:
Maybe here we can add a link to https://github.com/tableau/server-client-python/blob/master/docs/docs/filter-sort.md so that whoever wants to use filters can go there easily.